worlds/yachtdice/__init__.py

```diff
@@ -395,4 +395,18 @@ def fill_slot_data(self):
     def create_item(self, name: str) -> Item:
         item_data = item_table[name]
         item = YachtDiceItem(name, item_data.classification, item_data.code, self.player)
-        return item
\ No newline at end of file
+        return item
+
+    def collect(self, state: CollectionState, item: Item) -> bool:
+        change = super().collect(state, item)
+        if change and item.advancement:
```

Comment:
The more specific you can be here about "which items does the state actually need to be recalculated on", the faster this will be

Reply:
...except for the exact way I did it, which is just slower lol, `if change` is better, you already only get advancements in this function, my b